### PR TITLE
Add extra test cases

### PR DESCRIPTION
**Pull request by vinistock**
### Motivation

Add some new fixtures with some more uncommonly found syntax to make sure our requests can handle them.

### CODE REVIEW

File: test/fixtures/nested_block_arguments.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[[1, [2, 3]]].each do |one, (two, three)|
```
Comment (st0012) [inline test/fixtures/nested_block_arguments.rb:1]: Isn't this covered by `test/fixtures/double_nested_arguments.rb` as well?

**Review by st0012 — COMMENTED** [review]
Not saying we **should** do that, but if we want to find some edge cases to cover, [Ruby tricks snippets](https://github.com/tric) feels like a good source 😂  

[Example](https://github.com/tric/trick2022/blob/master/03-mame/entry.rb):

```rb
2022;"#

.chars} {puts'TRICK+2022'
        \  {         ;
#';$><< b 
   ?!
};#{s=' ' #           0 
! s[0]? (   b=$<.read  ;'
}            ub(      ,''}
';a=            ''<<32
b.lines      {puts( ?.. *(
b.size)   .gsub(/./) {
b.sub!(        /^#$`\K(\S)
             /x,a)?$1:a }
               .rstrip)}):
      ( [ 12,1,12,11].
cycle { | i | t = ( s *
10<<
10)*
10+            %(\e[A)*
10
10.             times{t[i*
    _1  ] =  'TRICK+2022'[
    _1  ]     };$><<t
            sleep 1})
               }"
```

File: test/fixtures/nested_block_arguments.rb (hunk 1 above)
Comment (vinistock) [reply test/fixtures/nested_block_arguments.rb]: You're right, it is. I was capturing a few interesting cases and added them sequentially as I found them.

**Comment by vinistock** [comment]
@st0012 that is some unholy piece of code. I'm hoping no one is adding code like that to their codebases 😂. That said, those trick snippets might contain some interesting examples that we can test.

**Review by Morriar — APPROVED** [review]
Very nice! ❤️
